beluga/include/beluga/algorithm/estimation.hpp

```diff
@@ -169,9 +170,46 @@ std::pair<Sophus::SE2<Scalar>, Sophus::Matrix3<Scalar>> estimate(Poses&& poses,
   return std::pair{estimated_pose, covariance_matrix};
 }
 
-/// Computes mean and covariance Returns a pair consisting of the estimated mean pose and its covariance.
+/// Computes mean and standard deviation of a range of weighted scalars.
 /**
- * Given a range of 2D poses, computes the estimated pose by averaging the translation
+ * Given a range of scalars, computes the scalar mean and standard deviation.
+ *
+ * \tparam Scalars A [sized range](https://en.cppreference.com/w/cpp/ranges/sized_range) type whose
+ *  value type is `std::vector<Scalar>`.
+ * \tparam Weights A [sized range](https://en.cppreference.com/w/cpp/ranges/sized_range) type whose
+ *  value type is `Scalar`.
+ * \tparam Scalar The scalar value type of the given range of Scalars.
+ * \param scalars Range of scalars.
+ * \param weights Range of weights.
+ * \return The estimated mean and its standard deviation.
+ */
+template <
+    class Scalars,
+    class Weights,
+    class Scalar = ranges::range_value_t<Scalars>,
+    typename = std::enable_if_t<std::is_arithmetic_v<Scalar>>>
+std::pair<Scalar, Scalar> estimate(Scalars&& scalars, Weights&& weights) {
+  auto weights_view = weights | ranges::views::common;
+  const auto weights_sum = ranges::accumulate(weights, 0.0, std::plus<>{});
+  auto normalized_weights_view =
+      weights_view | ranges::views::transform([weights_sum](const auto& weight) { return weight / weights_sum; });
+
+  const Scalar weighted_mean = std::transform_reduce(
+      scalars.begin(), scalars.end(), normalized_weights_view.begin(), 0.0, std::plus<>{},
+      [](const auto& scalar, const auto& weight) { return scalar * weight; });
+
+  const Scalar weighted_squared_deviations = std::transform_reduce(
```

Comment:
Yes, it is true that it is necessary to multiply by this factor, at the time I thought that it was disregarded since all the particles have even a minimum weight, but to generalize this equation it is true that it is necessary to add this factor, now I check it.